src/PasswordReset.ts

```diff
@@ -61,7 +64,27 @@ export default class PasswordReset {
     ): Promise<IRequestTokenResponse> {
         this.password = newPassword;
         this.logoutDevices = logoutDevices;
-        return this.client.requestPasswordEmailToken(emailAddress, this.clientSecret, 1).then((res) => {
+        this.sendAttempt++;
+        return this.client.requestPasswordEmailToken(emailAddress, this.clientSecret, this.sendAttempt).then((res) => {
+            this.sessionId = res.sid;
+            return res;
+        }, function(err) {
```

Comment:
we should be using arrow functions, and `.catch()` in new/modified code.

Reply:
Done